### PR TITLE
feat(deps): add compatibility with php 8.0

### PR DESCRIPTION
**Pull request by JellyBellyDev**
[feat/add-compatibility-with-php80 eb6f021] feat(deps): add compatibility with php 8.0

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "repositories": [
         {
             "type": "vcs",
-            "url": "https://github.com/madisoft/phpspec-data-provider-extension"
+            "url": "https://github.com/JellyBellyDev/phpspec-data-provider-extension"
```
Comment (AlessandroMinoccheri) [inline composer.json:31]: Why this change? What is the difference? php8 compatibility?

File: composer.json [hunk 2]
```diff
@@ -38,12 +38,12 @@
         "symfony/console": "^4.0|^5.0"
     },
     "require-dev": {
-        "behat/behat": "^3.7",
-        "phpspec/phpspec": "^6.1",
-        "friends-of-phpspec/phpspec-code-coverage": "^4.3",
-        "friendsofphp/php-cs-fixer": "^2.16",
-        "moave/phpspec-data-provider-extension": "dev-master",
-        "vimeo/psalm": "^3.12"
+        "behat/behat": "^3.8",
```
Comment (AlessandroMinoccheri) [inline composer.json:41]: It's so important this change? Behat support php8 from 3.8version?

File: composer.json [hunk 3]
```diff
@@ -55,7 +55,11 @@
         "php-cs-fixer": "bin/php-cs-fixer fix -v --ansi",
         "phpspec": "bin/phpspec run --format=pretty --ansi",
         "phpspec-coverage": "bin/phpspec run -f progress -c phpspec-coverage.yml --ansi",
-        "behat": "bin/behat"
+        "behat": "bin/behat",
+        "docker:build:php74": "docker build -t pugx/poser:php74 -f .circleci/images/php74/Dockerfile .",
```
Comment (AlessandroMinoccheri) [inline composer.json:59]: why docker build here?

File: docker-compose.yml [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
     build:
-      context: docker/php-fpm
-      network: host
+      context: docker/php74
     working_dir: /application
+    stdin_open: true
+    volumes:
+      - .:/application:cached
+
+  php80:
```
Comment (AlessandroMinoccheri) [inline docker-compose.yml:12]: 👍 👍 👍

File: composer.json (hunk 1 above)
Comment (JellyBellyDev) [reply composer.json:31]: Yep! I created the PR: https://github.com/madisoft/phpspec-data-provider-extension/pull/2/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R17  
In the meantime we test with my fork!  
Do you see any problems?

File: composer.json (hunk 2 above)
Comment (JellyBellyDev) [reply composer.json:41]: Exactly: https://packagist.org/packages/behat/behat#v3.8.0

File: composer.json (hunk 3 above)
Comment (JellyBellyDev) [reply composer.json:59]: for documentation, do you see any problems?

File: composer.json (hunk 1 above)
Comment (AlessandroMinoccheri) [reply composer.json:31]: I understand, ok for me 👍

File: composer.json (hunk 2 above)
Comment (AlessandroMinoccheri) [reply composer.json:41]: ok perfect 👍

File: composer.json (hunk 3 above)
Comment (AlessandroMinoccheri) [reply composer.json:59]: No problems, only to understand the meaning of those lines inside composer.json 👍

Comment (JellyBellyDev) [reply composer.json:59]: it's a practice I've often seen make in node projects in package.json and I think I'll try it! :)

**Comment by JellyBellyDev** [comment]
Ok now the pipeline works 💪

**Review by AlessandroMinoccheri — APPROVED** [review]
LGTM 👍
